core/includes/file.inc

```diff
@@ -514,8 +514,8 @@ function file_save_htaccess($directory, $private = TRUE, $force_overwrite = FALS
 
   $htaccess_lines = file_htaccess_lines($private);
 
-  // Write the .htaccess file.
-  if (file_put_contents($htaccess_path, $htaccess_lines)) {
+  // Write the .htaccess file. Suppress any PHP error and write to watchdog.
+  if (@file_put_contents($htaccess_path, $htaccess_lines)) {
```

Comment:
This change is needed because any PHP error/warning will cause tests to fail. We shouldn't be allowing a raw PHP error to be surfaced to end-users anyway. We already have watchdog logging in place in the `else {}` statement that follows this.